Description: Don't enclose IPs between double quotes in combined log
 This is a fix for upstream bug #7730, which restores the combined log
 format that was in place until version 14.0.0. See also

 https://twistedmatrix.com/trac/ticket/7730

 for more details.
Author: Free Ekanayaka <freee@debian.org>
Bug-Debian: https://bugs.debian.org/772629
Forwarded: yes
Last-Update: 2014-11-09

Index: b/twisted/web/http.py
===================================================================
--- a/twisted/web/http.py
+++ b/twisted/web/http.py
@@ -1911,7 +1911,7 @@ def combinedLogFormatter(timestamp, requ
     referrer = _escape(request.getHeader(b"referer") or b"-")
     agent = _escape(request.getHeader(b"user-agent") or b"-")
     line = (
-        u'"%(ip)s" - - %(timestamp)s "%(method)s %(uri)s %(protocol)s" '
+        u'%(ip)s - - %(timestamp)s "%(method)s %(uri)s %(protocol)s" '
         u'%(code)d %(length)s "%(referrer)s" "%(agent)s"' % dict(
             ip=_escape(request.getClientIP() or b"-"),
             timestamp=timestamp,
Index: b/twisted/web/test/test_web.py
===================================================================
--- a/twisted/web/test/test_web.py
+++ b/twisted/web/test/test_web.py
@@ -1056,7 +1056,7 @@ class AccessLogTestsMixin(object):
 
         self.assertEqual(
             # Client IP
-            b'"1.2.3.4" '
+            b'1.2.3.4 '
             # Some blanks we never fill in
             b'- - '
             # The current time (circa 1234567890)
@@ -1156,7 +1156,7 @@ class CombinedLogFormatterTests(unittest
 
         line = http.combinedLogFormatter(timestamp, request)
         self.assertEqual(
-            u'"evil x-forwarded-for \\x80" - - [13/Feb/2009:23:31:30 +0000] '
+            u'evil x-forwarded-for \\x80 - - [13/Feb/2009:23:31:30 +0000] '
             u'"POS\\x81 /dummy HTTP/1.0" 123 - "evil \\x83" "evil \\x84"',
             line)
 
@@ -1248,7 +1248,7 @@ class LogEscapingTests(unittest.TestCase
         self.site._logDateTime = "[%02d/%3s/%4d:%02d:%02d:%02d +0000]" % (
             25, 'Oct', 2004, 12, 31, 59)
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HTTP/1.0" 123 - "-" "-"\n')
 
 
@@ -1260,7 +1260,7 @@ class LogEscapingTests(unittest.TestCase
             25, 'Oct', 2004, 12, 31, 59)
         self.request.method = b'G"T'
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"G\\"T /dummy HTTP/1.0" 123 - "-" "-"\n')
 
 
@@ -1272,7 +1272,7 @@ class LogEscapingTests(unittest.TestCase
             25, 'Oct', 2004, 12, 31, 59)
         self.request.uri = b'/dummy"withquote'
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy\\"withquote HTTP/1.0" 123 - "-" "-"\n')
 
 
@@ -1284,7 +1284,7 @@ class LogEscapingTests(unittest.TestCase
             25, 'Oct', 2004, 12, 31, 59)
         self.request.clientproto = b'HT"P/1.0'
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HT\\"P/1.0" 123 - "-" "-"\n')
 
 
@@ -1299,7 +1299,7 @@ class LogEscapingTests(unittest.TestCase
             b'referer',
             b'http://malicious" ".website.invalid')
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HTTP/1.0" 123 - '
             b'"http://malicious\\" \\".website.invalid" "-"\n')
 
@@ -1314,7 +1314,7 @@ class LogEscapingTests(unittest.TestCase
         self.request.requestHeaders.addRawHeader(b'user-agent',
                                                  b'Malicious Web" Evil')
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HTTP/1.0" 123 - "-" "Malicious Web\\" Evil"\n')
 
 
